railties/test/commands/server_test.rb

```diff
@@ -118,4 +118,17 @@ def test_default_options
       assert_equal old_default_options, server.default_options
     end
   end
+
+  def test_restart_command_contains_customized_options
+    args = ["-p", "4567"]
+    ARGV.replace args
```

Comment:
Is this inline mutation a good idea? Wouldn't at least be better to capture the old ARGV value and reset in ensure.